client/me/purchases/manage-purchase/payment-info-block.tsx

```diff
@@ -31,6 +31,16 @@ export default function PaymentInfoBlock( {
 			card.meta?.find( ( meta ) => meta.meta_key === 'is_backup' )?.meta_value
 	);
 
+	if ( ! isRechargeable( purchase ) && hasPaymentMethod( purchase ) ) {
+		return (
+			<PaymentInfoBlockWrapper>
```

Comment:
This component has so many possible outputs that we added [unit tests for it here](https://github.com/Automattic/wp-calypso/blob/c8b35df0d209e20a9b6365468cbc894fad320624/client/me/purchases/manage-purchase/test/payment-info-block.js#L318). Could we add some unit tests for this change as well?

Reply:
Good Catch, let me add a Unit test for this.

Reply:
@sirbrillig just sent a commit fixing the failing test when using a non rechargeable payment method.